rag/tests/test_research.py

```
@@ -0,0 +1,85 @@
+
+def test_run_deep_research_raises_fallback_error_when_both_fail(monkeypatch):
+    monkeypatch.setenv("OPENAI_API_KEY", "sk-test")
+    mock_client = MagicMock()
+    primary_err = Exception("primary failed")
+    fallback_err = Exception("fallback failed")
+    mock_client.responses.create.side_effect = [primary_err, fallback_err]
+
+    with patch("main.OpenAI", return_value=mock_client):
+        try:
+            main.run_deep_research("テスト社", "エンジニア")
+            assert False, "例外が送出されるはず"
```

Ruff (0.16.1)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)